File: beecrowd/tentados/1025-ondeEstaMarmore.py
def merge_sort(arr):
    if len(arr) > 1:
        mid = len(arr) // 2
        L = arr[:mid]
        R = arr[mid:]

        merge_sort(L)
        merge_sort(R)

        i = j = k = 0

        while i < len(L) and j < len(R):
            if L[i] < R[j]:
                arr[k] = L[i]
                i += 1
            else:
                arr[k] = R[j]
                j += 1
            k += 1

        while i < len(L):
            arr[k] = L[i]
            i += 1
            k += 1

        while j < len(R):
            arr[k] = R[j]
            j += 1
            k += 1


contador = 1
while True:
    numMarmore, numBuscas = [int(x) for x in input().split()]
    if numMarmore == 0:
        break

    vetor = []
    for i in range(numMarmore):
        vetor.append(int(input()))
    merge_sort(vetor)

    print(f"CASE# {contador}:")
    contador+=1
    while numBuscas:
        busca = int(input())
        for i, x in enumerate(vetor):
            if i == busca:
                print(f"{busca} found at {x+1}")
                break
        else:
            print(f"{busca} not found")
        numBuscas-=1

File: beecrowd/resolvidos/1025-ondeEstaMarmore.cpp
#include <bits/stdc++.h>
using namespace std;

void merge(std::vector<int>& arr, int l, int m, int r) {
    int n1 = m - l + 1;
    int n2 = r - m;
    std::vector<int> L(n1), R(n2);

    for (int i = 0; i < n1; i++) {
        L[i] = arr[l + i];
    }
    for (int j = 0; j < n2; j++) {
        R[j] = arr[m + 1 + j];
    }

    int i = 0, j = 0, k = l;
    while (i < n1 && j < n2) {
        if (L[i] <= R[j]) {
            arr[k] = L[i];
            i++;
        }
        else {
            arr[k] = R[j];
            j++;
        }
        k++;
    }

    while (i < n1) {
        arr[k] = L[i];
        i++;
        k++;
    }

    while (j < n2) {
        arr[k] = R[j];
        j++;
        k++;
    }
}

void mergeSort(std::vector<int>& arr, int l, int r) {
    if (l < r) {
        int m = l + (r - l) / 2;
        mergeSort(arr, l, m);
        mergeSort(arr, m + 1, r);
        merge(arr, l, m, r);
    }
}

int main(void){
    int numMarmore, numTestes, num, busca, contador=1;
    while(cin >> numMarmore >> numTestes && numMarmore){
        vector<int> vetor;
        for(int x=0; x<numMarmore; x++){
            cin>>num;
            vetor.push_back(num);
        }
        
        mergeSort(vetor, 0, numMarmore-1);

        cout << "CASE# " << contador++ << ":"<< endl;
        while(numTestes--){
            cin >> busca;
            
            int achou=-1;
            for(int x=0; x<vetor.size();x++){
                if(vetor[x]==busca) {
                    achou = x;
                    break;
                }
            }

            if(achou==-1) cout << busca << " not found";
            else cout << busca << " found at " << achou+1;
            cout << endl;
        }
    }
    return 0;
}